exp/views/user.py

```diff
@@ -70,10 +76,37 @@ class ResearcherListView(LoginRequiredMixin, generic.ListView):
     queryset = User.objects.filter(demographics__isnull=True)
     model = User
 
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_org_read or request.user.is_superuser:
+            return super().dispatch(request, *args, **kwargs)
+        else:
+            raise PermissionDenied # HTTP 403
+
+
     def get_queryset(self):
         qs = super(ResearcherListView, self).get_queryset()
         # TODO this should probably use permissions eventually, just to be safe
-        return qs.filter(organization=self.request.user.organization)
+        queryset = qs.filter(organization=self.request.user.organization)
+        match = self.request.GET.get('match')
+        if match:
+            queryset = queryset.filter(Q(family_name__icontains=match) | Q(given_name__icontains=match)| Q(middle_name__icontains=match))
+        sort = self.request.GET.get('sort')
+        if sort:
+            if 'family_name' in sort:
+                queryset = queryset.order_by(Lower('family_name').desc()) if '-' in sort else queryset.order_by(Lower('family_name').asc())
+        return queryset
+
+    def post(self, request, *args, **kwargs):
+        retval = super(ResearcherListView, self).get(request, *args, **kwargs)
```

Comment:
This is a ListView so there isn't a post method on it.  It's a list view, but the mockups have the delete button on this page. If I add a deleteview to this, then I get  `object has no attribute 'object'` error. i needed to return something here.